spec/models/my_academics/my_academic_roles_spec.rb

```diff
@@ -12,80 +27,87 @@
       {'term_id'=>'2172', 'acad_career'=>'GRAD', 'acad_program'=>'UCNR', 'acad_plan'=>'04606U'},
     ]
   end
-  let(:described_class_instance) { described_class.new(random_id) }
+
+  before do
+    allow_any_instance_of(Berkeley::Term).to receive(:campus_solutions_id).and_return('2172')
+    allow(User::Current).to receive(:new).with(uid).and_return(current_user)
+    allow(MyAcademics::MyTermCpp).to receive(:new).with(uid).and_return(my_term_cpp)
+  end
 
   describe '#get_feed_internal' do
-    subject { described_class_instance.get_feed_internal }
+    let(:result) { subject.get_feed_internal }
```

Comment:
Isn't the subject of this block the result?

Reply:
@developish - I really haven't like how people override the subject to be the method call itself rather than the instance of the class. They're doing it just so that they can use the shorter assertion syntax like `it { should be false }`, but then in those contexts you can't stub anything on the instance because `subject` has been redefined.

I think it's bad form. I think the `subject` in the entire document should be the instance.

I know this flows from the way I'm trying to isolate the methods in their own respective `describe` blocks. Perhaps my approach is misguided and there's a better way. Maybe all our specs should integrate the methods within a class instead of testing them isolated from each other, and that when some class gets too complicated it should be abstracted. Perhaps all this dysfunction flows from making everything a machine that processes some indefinite Hash, and not a properly defined Ruby object that IS something, rather than DOES something.

Reply:
I buy the "I think the subject in the entire document should be the instance." argument 👍